### PR TITLE
Deprecate compression options

### PR DESCRIPTION
**Pull request by echarlaix**
In this PR we deprecate `compression_option` and `compression_ratio` during export, in favor of `ov_config` which will handle all quantization parameters (`group_size`, `ratio`, sym, ...) through `quantization_config` and `dtype` (for fp16 export)

Also add an `quantization_config` argument for all `OVModels` to enable 8bit and 4bit quantization
```python
model = OVModelForXxx.from_pretrained(model_id, quantization_config={"bits" : 4})
```

### CODE REVIEW

File: docs/source/inference.mdx [hunk 1]
```diff
@@ -96,15 +98,23 @@ tokenizer.save_pretrained(save_directory)
...
-This can also be done when loading your model by setting the `load_in_8bit` argument when calling the `from_pretrained()` method.
+By default the quantization scheme will be [assymmetric](https://github.com/openvinotoolkit/nncf/blob/develop/docs/compression_algorithms/Quantization.md#asymmetric-quantization), to make it [symmetric](https://github.com/openvinotoolkit/nncf/blob/develop/docs/compression_algorithms/Quantization.md#symmetric-quantization) you can add `--sym`.
+
+For INT4 quantization you can also specify the following arguments :
+* The `--group-size` parameter will define the group size to use for quantization, `-1` it will results in per-column quantization.
+* The `--ratio` CLI parameter controls the ratio between 4-bit and 8-bit quantization. If set to 0.9, it means that 90% of the layers will be quantized to `int4` while 10% will be quantized to `int8`.
+
+Smaller `group_size` and `ratio` of usually improve accuracy at the sacrifice of the model size and inference latency.
+
+You can also apply apply 8-bit quantization on your model's weight when loading your model by setting the `load_in_8bit=True` argument when calling the `from_pretrained()` method.
```
Comment (AlexKoff88) [inline docs/source/inference.mdx:117]:
```suggestion
You can also apply 8-bit quantization on your model's weight when loading your model by setting the `load_in_8bit=True` argument when calling the `from_pretrained()` method.
```

**Comment by AlexKoff88** [comment]
Thanks, @echarlaix. It looks great!

Here are a couple of comments from my side:
- How are the default 4-bit configs supposed to work on the API level? Like `model=OVModelForCausalLM.from_pretrained(model_id, ov_config=dict(dtype="int4"))`?
- It is worth adding a test that checks this flow on the API level. We can do it with dummy models and patching `_DEFAULT_4BIT_CONFIGS` with `unittest.mock.patch()`.

**Comment by echarlaix** [comment]
>    How are the default 4-bit configs supposed to work on the API level? Like `model=OVModelForCausalLM.from_pretrained(model_id, ov_config=dict(dtype="int4"))`?

Currently the user needs to specify the `quantization_config` :
```python
model = OVModelForSequenceClassification.from_pretrained("/tmp/test", quantization_config={"bits" : 4})
```
do you think `ov_config` should be added as an argument ?
 
>    It is worth adding a test that checks this flow on the API level. We can do it with dummy models and patching `_DEFAULT_4BIT_CONFIGS` with `unittest.mock.patch()`.

`_DEFAULT_4BIT_CONFIGS` is used in the CLI so are you thinking about adding a test in [test_exporters_cli.py](https://github.com/huggingface/optimum-intel/blob/main/tests/openvino/test_exporters_cli.py) to see if the correct quantization config is given for a dummy model (after patching `_DEFAULT_4BIT_CONFIGS`) ? We would also need to add the saving of the `OVConfig` after export (was planning to add it in a following PR but can add it in this one if needed)

**Comment by AlexKoff88** [comment]
> > How are the default 4-bit configs supposed to work on the API level? Like `model=OVModelForCausalLM.from_pretrained(model_id, ov_config=dict(dtype="int4"))`?
> 
> Currently the user needs to specify the `quantization_config` :
> 
> ```python
> model = OVModelForSequenceClassification.from_pretrained("/tmp/test", quantization_config={"bits" : 4})
> ```
> 
> do you think `ov_config` should be added as an argument ? (I think we should have either `quantization_config` or `ov_config`)
> 
> > It is worth adding a test that checks this flow on the API level. We can do it with dummy models and patching `_DEFAULT_4BIT_CONFIGS` with `unittest.mock.patch()`.
> 
> `_DEFAULT_4BIT_CONFIGS` is used in the CLI so are you thinking about adding a test in [test_exporters_cli.py](https://github.com/huggingface/optimum-intel/blob/main/tests/openvino/test_exporters_cli.py) to see if the correct quantization config is given for a dummy model (after patching `_DEFAULT_4BIT_CONFIGS`) ? We would also need to add the saving of the `OVConfig` after export (was planning to add it in a following PR but can add it in this one if needed)

Now, I see that `_DEFAULT_4BIT_CONFIGS` are used only in CLI. On the API level, we still have a warning message about the recommended config. Can we align the CLI and API behaviors by any chance?

I am ok with adding the test in the follow-up PR.

File: optimum/intel/openvino/modeling_decoder.py [hunk 2]
```diff
@@ -575,11 +576,18 @@ def _from_pretrained(
             local_files_only=local_files_only,
         )
 
+        # Give default quantization config if not provided and load_in_8bit=True
+        if load_in_8bit:
+            quantization_config = quantization_config or {"bits": 8}
+
         if isinstance(quantization_config, dict):
+            if quantization_config == {"bits": 4} and config.name_or_path in _DEFAULT_4BIT_CONFIGS:
```
Comment (AlexKoff88) [inline optimum/intel/openvino/modeling_decoder.py:584]: thanks, @echarlaix. I'm fine with merging this if it works.

**Review by AlexKoff88 — APPROVED** [review]

File: optimum/intel/openvino/modeling_decoder.py (hunk 2 above)
Comment (echarlaix) [reply optimum/intel/openvino/modeling_decoder.py:584]: perfect, will add the config saving + test mentioned above in a following PR

**Pull request merged**